src/client/src/pages/settings/components/editorSettingList/mapSettings.jsx

```diff
@@ -1,44 +1,199 @@
+            }}>
+            <Highlighter searchWords={[state.searchWms]} textToHighlight={layer.Abstract} />
+          </div>
+        </div>
+      ) : null,
+    );
+  }
+
+  function getWmtsList() {
+    console.log(state.wmts);
+    return state.wmts.Contents.Layer.map((layer, idx) => {
+      return state.searchWmts === "" ||
+        (Object.prototype.hasOwnProperty.call(layer, "Title") &&
+          layer.Title.toLowerCase().search(state.searchWmts) >= 0) ||
+        (Object.prototype.hasOwnProperty.call(layer, "Abstract") &&
+          layer.Abstract.toLowerCase().search(state.searchWmts) >= 0) ||
+        (Object.prototype.hasOwnProperty.call(layer, "Identifier") &&
+          layer.Identifier.toLowerCase().search(state.searchWmts) >= 0) ? (
+        <div
+          className="selectable unselectable"
+          key={"wmts-list-" + idx}
+          style={{
+            padding: "0.5em",
+          }}>
+          <Stack
+            data-cy="wmts-list-box"
+            direction="row"
+            alignItems="center"
+            sx={{
+              fontWeight: "bold",
+            }}>
+            <div
+              style={{
+                flex: 1,
+              }}>
+              <Highlighter searchWords={[state.searchWmts]} textToHighlight={layer.Title} />
+            </div>
+            <div>{getIconButton(layer, "WMTS")}</div>
+          </Stack>
+          <div
+            style={{
+              color: "#787878",
+              fontSize: "0.8em",
+            }}>
+            <Highlighter searchWords={[state.searchWmts]} textToHighlight={layer.Identifier} />
+          </div>
+          <div
+            style={{
+              fontSize: "0.8em",
+            }}>
+            <Highlighter searchWords={[state.searchWmts]} textToHighlight={layer.Abstract} />
+          </div>
+        </div>
+      ) : null;
+    });
+  }
+
+  function fetchCapabilitiesForService() {
+    const isWms = setting.selectedWMS.startsWith("https://wms");
+    const languageParam = `${isWms ? "&lang=" : "?lang="}${i18n.language}`;
+    fetch(setting.selectedWMS + languageParam).then(response => {
+      response.text().then(data => {
+        // Check if WMS or WMTS
+        if (/<(WMT_MS_Capabilities|WMS_Capabilities)/.test(data)) {
+          const wms = new WMSCapabilities().read(data);
+          setState({
+            ...state,
+            wmsFetch: false,
+            wms: wms,
+            wmts: null,
+          });
+        } else if (/<Capabilities/.test(data)) {
+          const wmts = new WMTSCapabilities().read(data);
+          setState({
+            ...state,
+            wmsFetch: false,
+            wms: null,
+            wmts: wmts,
+          });
+        } else {
+          setState({
+            ...state,
+            wmsFetch: false,
+            wms: null,
+            wmts: null,
+          });
+          showAlert("Sorry, only Web Map Services (WMS) and " + "Web Map Tile Service (WMTS) are supported", "error");
```

Comment:
Q: internationalization?
PP:
```suggestion
          showAlert("Sorry, only Web Map Services (WMS) and " + "Web Map Tile Services (WMTS) are supported", "error");
```